deployment/run.sh

```diff
@@ -83,290 +23,11 @@ if [ -n "${WRITE_KUBECONFIG}" ]; then
-	# generate common node key for the SearchGuard plugin
-	openssl rand 16 | openssl enc -aes-128-cbc -nosalt -out $dir/searchguard_node_key.key -pass pass:pass
-
-	# generate proxy session
-	cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 200 | head -n 1 > "$dir/session-secret"
-	# generate oauth client secret
-	cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1 > "$dir/oauth-secret"
-
-	# (re)generate secrets
-	echo "Deleting existing secrets"
-	oc delete secret logging-fluentd logging-elasticsearch logging-kibana logging-kibana-proxy logging-kibana-ops-proxy logging-curator logging-curator-ops || :
-
-	echo "Creating secrets"
-	oc secrets new logging-elasticsearch \
-	    key=$dir/keystore.jks truststore=$dir/truststore.jks \
-	    searchguard.key=$dir/searchguard_node_key.key
-	oc secrets new logging-kibana \
-	    ca=$dir/ca.crt \
-	    key=$dir/${kibana_user}.key cert=$dir/${kibana_user}.crt
-	oc secrets new logging-kibana-proxy \
-	    oauth-secret=$dir/oauth-secret \
-	    session-secret=$dir/session-secret \
-	    server-key=$dir/kibana.key \
-	    server-cert=$dir/kibana.crt \
-	    server-tls.json=$dir/server-tls.json
-	oc secrets new logging-kibana-ops-proxy \
-	    oauth-secret=$dir/oauth-secret \
-	    session-secret=$dir/session-secret \
-	    server-key=$dir/kibana-ops.key \
-	    server-cert=$dir/kibana-ops.crt \
-	    server-tls.json=$dir/server-tls.json
-	oc secrets new logging-fluentd \
-	    ca=$dir/ca.crt \
-	    key=$dir/${fluentd_user}.key cert=$dir/${fluentd_user}.crt
-	oc secrets new logging-curator \
-	    ca=$dir/ca.crt \
-	    key=$dir/${curator_user}.key cert=$dir/${curator_user}.crt
-	oc secrets new logging-curator-ops \
-	    ca=$dir/ca.crt \
-	    key=$dir/${curator_user}.key cert=$dir/${curator_user}.crt
-
-fi # supporting infrastructure
-
-# (re)generate templates needed
-echo "Creating templates"
-oc delete template --selector logging-infra=curator
-oc delete template --selector logging-infra=kibana
-oc delete template --selector logging-infra=fluentd
-oc delete template --selector logging-infra=elasticsearch
-
-es_params=$(join , \
-	ES_CLUSTER_NAME=es \
-	ES_INSTANCE_RAM=${es_instance_ram} \
-	ES_NODE_QUORUM=${es_node_quorum} \
-	ES_RECOVER_AFTER_NODES=${es_recover_after_nodes} \
-	ES_RECOVER_EXPECTED_NODES=${es_recover_expected_nodes} \
-	ES_RECOVER_AFTER_TIME=${es_recover_after_time} \
-	)
-
-es_ops_params=$(join , \
-	ES_CLUSTER_NAME=es-ops \
-	ES_INSTANCE_RAM=${es_ops_instance_ram} \
-	ES_NODE_QUORUM=${es_ops_node_quorum} \
-	ES_RECOVER_AFTER_NODES=${es_ops_recover_after_nodes} \
-	ES_RECOVER_EXPECTED_NODES=${es_ops_recover_expected_nodes} \
-	ES_RECOVER_AFTER_TIME=${es_ops_recover_after_time} \
-	)
-
-if [[ -n "${ES_NODESELECTOR}" ]]; then
-	sed "/serviceAccountName/ i\
-\          ${es_nodeselector}" templates/es.yaml | oc process -v "${es_params}" -f - | oc create -f -
+if [ "${mode}" == "install" ]; then
```

Comment:
case "${mode}" in
     install)
      ;;
    ...

minor thing but i expect this list to grow...